src/main/perl/Partition.pm

```diff
@@ -627,32 +623,50 @@ sub create_pre_ks
     print <<EOF;
 if ! grep -q '$self->{devname}\$' /proc/partitions
 then
-    echo "Creating partition $self->{devname}"
-    prev=\`parted $disk -s u MiB p |awk '\$1==$prev_n {print \$5=="$extended_txt" ? \$2:\$3}'\`
+    sectsize=\$(blockdev --getss $disk)
+    if ((sectsize == 0)); then
+        sectsize=512
+    fi
+    opt_io=\$(blockdev --getioopt $disk)
+    if ((opt_io == 0)) || ((1024 * 1024 % opt_io == 0)); then
+        opt_io=\$((1024 * 1024))
+    fi
+    align_sec=\$((opt_io / sectsize))
+EOF
 
-    if [ -z \$prev ]
-    then
-        prev=1
+    if (defined($offset)) {
+        print <<EOF;
+    offset_sec=\$(($offset * (1024 * 1024 / sectsize)))
 EOF
+    } else {
+        print <<EOF;
+    offset=\$(blockdev --getalignoff $disk)
+    offset_sec=\$((offset / sectsize))
```

Comment:
`offset` -> `\$offset`; same for sectsize

Reply:
heh, this actually works. sorry for doubting your bash skills :wink:

Reply:
Actually, having to do less quoting pushed me in the direction of using `(( ... ))` instead of `[ ... ]` where possible :-)